server/src/graphql/mutations/createAppGithub.ts

```diff
@@ -32,43 +34,42 @@ export const createAppGithub: MutationResolvers['createAppGithub'] = async (
     },
   });
 
+  const user = await prisma.user.findUnique({
+    where: {
+      id: userId,
+    },
+  });
+
   const isAppNameTaken = !!apps[0];
 
   if (isAppNameTaken) {
     throw new Error('App name already taken');
   }
 
-  const octokit = new Octokit({});
-
-  const repoData = getRepoData(input.gitRepoUrl);
-
-  const repo = await octokit.repos.get({
-    owner: repoData.owner,
-    repo: repoData.repoName,
+  const auth = createAppAuth({
+    appId: config.githubAppId,
+    privateKey: config.githubAppPem,
+    clientId: config.githubAppClientSecret,
+    clientSecret: config.githubAppClientSecret,
   });
 
-  if (!repo) {
-    throw new Error(
-      `No repository found for ${repoData.owner} with name ${repoData.repoName}`
-    );
-  }
-
-  const branch = await octokit.repos.getBranch({
```

Comment:
@Akirtovskis I think we should still verify that the branch does exist

Reply:
We should also check that the repoId is valid

Reply:
@pradel reason I removed these checks is that on the client side we are fetching repo and branch data from the Github app instead of entering manually so the validation becomes redundant.

Do we still need to validate?

Reply:
The rule is to never trust the client as I could manually call this endpoint with some fake data.

Reply:
Ok, will add this back then.